db/commitlog/commitlog.cc

```diff
@@ -3135,6 +3380,88 @@ db::commitlog::read_log_file(sstring filename, sstring pfx, commit_load_reader_f
+
+                        auto in = buf.get_istream();
+                        auto nsize = read<size_t>(in);
+                        sstring filename(nsize, '\0');
+                        in.read_to(nsize, const_cast<char*>(filename.c_str()));
+                        auto off = read<uint64_t>(in);
+                        auto size = read<size_t>(in);
+                        // See oversized_alloc. Only first entry for a given file name
+                        // is written into the array. Just reuse previous one if so.
+                        if (filename.empty() && !tmp.empty()) {
+                            filename = std::get<sstring>(tmp.back());
+                        }
+                        tmp.emplace_back(std::move(filename), off, size, rp);
+                        co_return;
+                    };
+
+                    while (pos < end) {
+                        co_await read_entry(add_entry);
+                        if (failed) {
+                            break;
+                        }
+                    }
+
+                    auto i = tmp.begin();
+                    auto e = tmp.end();
+
+                    // Now process the metadata. Note: this code
+                    // allows for an array with more than one external
+                    // file used. Currently that is not used, but why not...
+                    while (i != e) {
+                        auto j = i;
+                        auto filename = std::get<sstring>(*i); 
+                        auto off = std::get<1>(*i);
+
+                        while (++j != e) {
+                            // find partition endpoint (not used yet, will find end, but..)
+                            if (filename != std::get<sstring>(*j)) {
+                                break;
+                            }
+                        }
+
+                        auto produce_entry = [&](buffer_and_replay_position buf_rp) -> future<> {
+                            auto&& rp = buf_rp.position;
+                            // Check if we have data loss in the external file...
+                            while (i != j && std::get<1>(*i) < rp.pos) {
+                                clogger.warn("Segment entry at {}:{} missing", filename, std::get<replay_position>(*i));
+                                ++i;
+                            }
+                            // Now we have the metadata for the entry.
+                            if (i != j && std::get<1>(*i) == rp.pos) {
+                                if (std::get<2>(*i) != buf_rp.buffer.size_bytes()) {
+                                    throw std::runtime_error("Mismatched size in external entry");
+                                }
+                                // pretend the replay position is the reference location
+                                buf_rp.position = std::get<replay_position>(*i);
+                                co_await this->func(std::move(buf_rp));
```

Comment:
Shouldn't this be just `func`?

Reply:
local var has same name. Here we want to explicitly call the original (caller provided) callback.

Reply:
But you are right, here we can just do "func".